src/react-apps/applications/runtime/src/components/base/FileUploadComponent.tsx

```diff
@@ -1,45 +1,332 @@
+      ...state,
+      attachments: props.attachments,
+    };
   }
 
-  public render() {
-    const descriptionKey = this.props.component.textResourceBindings.title;
+  constructor(props: IFileUploadProps, state: IFileUploadState) {
+    super(props, state);
+    this.state = {
+      attachments: props.attachments || [],
+      validations: [],
+      ...state,
+    };
+  }
+
+  public onDrop = (acceptedFiles: File[], rejectedFiles: File[]) => {
+    const newFiles: IAttachment[] = [];
+    const fileType = this.props.id; // component id used as filetype identifier for now, see issue #1364
+    acceptedFiles.forEach((file: File) => {
+      if (this.state.attachments.length < this.props.component.maxNumberOfAttachments) {
+        const tmpId: string = uuid();
+        newFiles.push({ name: file.name, size: file.size, uploaded: false, id: tmpId, deleting: false });
+        FormFillerActionDispatchers.uploadAttachment(file, fileType, tmpId, this.props.id);
+      }
+    });
+    let newValidationMessages: string[];
+    if (rejectedFiles.length > 0) {
+      newValidationMessages = [];
+      rejectedFiles.forEach((file) => {
+        if (file.size > (this.props.component.maxFileSizeInMB * 1000 * 1000)) {
+          newValidationMessages.push(
+            file.name +
+            getLanguageFromKey('form_filler.file_uploader_validation_error_file_size', this.props.language));
+        } else {
+          newValidationMessages.push(
+            getLanguageFromKey('form_filler.file_uploader_validation_error_general_1', this.props.language) + ' ' +
+            file.name + ' ' +
+            getLanguageFromKey('form_filler.file_uploader_validation_error_general_2', this.props.language));
+        }
+      });
+    }
+    const showFileUpload = (this.props.component.displayMode === 'simple') ? false :
+      (this.props.attachments.length < this.props.component.maxFileSizeInMB);
+    this.setState({
+      attachments: this.state.attachments.concat(newFiles),
+      // if simple mode, we should hide list on each drop
+      showFileUpload,
+      validations: newValidationMessages || [],
+    });
+  }
+
+  public handleDeleteFile = (index: number) => {
+    const attachmentToDelete = this.state.attachments[index];
+    const fileType = this.props.id; // component id used as filetype identifier for now, see issue #1364
+    attachmentToDelete.deleting = true;
+    if (attachmentToDelete.uploaded) {
+      FormFillerActionDispatchers.deleteAttachment(attachmentToDelete, fileType, this.props.id);
+    }
+    const newList = this.state.attachments.slice();
+    newList[index] = attachmentToDelete;
+    this.setState({
+      attachments: newList,
+    });
+  }
+
+  public getComponentValidations = (): IComponentValidations => {
+    const { validations } = this.state;
+    let { validationMessages } = this.props;
+    if (!validationMessages || !validationMessages.simpleBinding) {
+      validationMessages = {
+        ['simpleBinding']: {
```

Comment:
What is this 'simpleBinding'?

Reply:
A default key for validationmessages/datamodelbindings. The idea is to be able to use keys to map databindings and validation-messages to specific fields in more complex components.